src/main/java/org/apache/ibatis/type/TypeHandlerRegistry.java

```diff
@@ -241,7 +241,8 @@ public boolean hasTypeHandler(TypeReference<?> javaTypeReference, JdbcType jdbcT
     }
     if (jdbcHandlerMap == null && type instanceof Class) {
       Class<?> clazz = (Class<?>) type;
-      if (clazz.isEnum()) {
+      if (clazz.isEnum() || (clazz.isAnonymousClass() && clazz.getEnclosingClass().isEnum())) {
```

Comment:
I think that the following check works, but could you test it please? 🙏😅

```suggestion
      if (Enum.class.isAssignableFrom(clazz)) {
```

Reply:
@harawata Thanks!! I will update ASAP.

Reply:
fixed via f7ed824840a0c1d82b6887b26221b0327243eb39